datadog/resource_datadog_synthetics_test_.go

```diff
@@ -335,6 +335,37 @@ func syntheticsTestOptionsList() *schema.Schema {
 					Type:     schema.TypeInt,
 					Optional: true,
 				},
+				"monitor_options": {
+					Type:     schema.TypeMap,
```

Comment:
It's probably better to not use TypeMap and use the same trick with TypeList and MaxItems 1. Otherwise your schema isn't really useful, and you need to do all the type casting below.